faust/transport/conductor.py

```diff
@@ -413,7 +413,7 @@ def add(self, topic: TopicT) -> None:
 
     def _topic_contain_unsubscribed_topics(self, topic: TopicT) -> bool:
         index = self._topic_name_index
-        return bool(index and any(t not in index for t in topic.topics))
+        return bool(any(t not in index for t in topic.topics))
```

Comment:
Not sure how I feel about removing the check for `index`, I need to think about edge cases where this could cause things to break.

Reply:
That literally says: "only check that that the topics are in `index` if `index` is non-empty"... the only case that I can think of where you might want to have that behaviour is if you want to be sure that there's an assignor subscription, but that doesn't make much sense to me as there's surely a more explicit way of doing that?

Reply:
> the only case that I can think of where you might want to have that behaviour is if you want to be sure that there's an assignor subscription

Good point. I looked through all the code we have and it seems we don't need to worry about that, I was being overly cautious. I also reviewed the git blame and this method hasn't gotten attention since 2018, when the project was undergoing more changes. Looks like Ask included the `index` check to support the `Mock` return_value.